e2e/testcases/multi_sync_test.go

```diff
@@ -771,17 +778,31 @@ func roleHasRules(wantRules []rbacv1.PolicyRule) nomostest.Predicate {
 
 // waitInParallel executes multiple functions in parallel until they all return,
 // which allows waiting for multiple conditions simultaniously, without
-// requiring them all to be true at the same time.
+// requiring them all to be successful at the same time.
 // This improves testing speed and consistency.
-func waitInParallel(fns ...func()) {
+// Returns a compound error, if any occured.
+func waitInParallel(fns ...func() error) error {
 	var wg sync.WaitGroup
+	var errs []error
+	errCh := make(chan error)
 	for i := range fns {
 		fn := fns[i]
 		wg.Add(1)
 		go func() {
-			defer wg.Done()
-			fn()
+			defer func() {
+				wg.Done()
+			}()
+			errCh <- fn()
 		}()
 	}
-	wg.Wait()
+	go func() {
+		wg.Wait()
+		close(errCh)
```

Comment:
minor nit: is it safer to `defer close(errCh)` before the call to `wg.Wait()`?

Reply:
I honestly have no idea. It depends if Wait panics.
That said, we're probably more at risk of a deadlock than a panic from the stdlib.

Reply:
Added a defer to be defensive, but i dont actually think it matters. 

I guess if a panic is caught by the test runner you'd want the other goroutines to exit cleanly too...